test/rhsmlib/dbus/test_facts.py

```diff
@@ -10,40 +10,33 @@
 # Red Hat trademarks are not licensed under GPLv2. No permission is
 # granted to use or replicate Red Hat trademarks that are incorporated
 # in this software or its documentation.
-import dbus
-import dbus.exceptions
+import mock
 
 from rhsmlib.dbus.facts.base import AllFacts
-from rhsmlib.dbus.facts import constants
 
-from test.rhsmlib.base import DBusObjectTest
+from test.rhsmlib.base import DBusServerStubProvider
 
-from test import subman_marker_dbus
 
+class TestFactsDBusObject(DBusServerStubProvider):
+    dbus_class = AllFacts
+    dbus_class_kwargs = {}
 
-@subman_marker_dbus
-class TestFactsDBusObject(DBusObjectTest):
-    def setUp(self):
-        super(TestFactsDBusObject, self).setUp()
-        self.proxy = self.proxy_for(AllFacts.default_dbus_path)
-        self.interface = dbus.Interface(self.proxy, constants.FACTS_DBUS_INTERFACE)
+    @classmethod
+    def setUpClass(cls) -> None:
+        # Do not try to use system virt-what
+        get_virt_info_patch = mock.patch(
+            "rhsmlib.facts.virt.VirtWhatCollector.get_virt_info",
+            name="get_virt_info",
+        )
+        cls.patches["get_virt_info"] = get_virt_info_patch.start()
+        cls.addClassCleanup(get_virt_info_patch.stop)
 
-    def dbus_objects(self):
-        return [AllFacts]
+        super().setUpClass()
 
-    def bus_name(self):
-        return constants.FACTS_DBUS_NAME
+    def setUp(self) -> None:
```

Comment:
this needs to call `setUp()` of the parent class (`DBusServerStubProvider`), so in case we add one implementation in the future it will be used as expected